### PR TITLE
Update dependencies

### PR DESCRIPTION
**Pull request by iandonovan**
Prompted by #109, I've updated our dependencies within the limits of the gemspec / gemfile. Here's the gems that are left behind:

```
Outdated gems included in the bundle:
  * database_cleaner (1.4.0 > 1.0.1) Gemfile specifies "= 1.0.1"
  * jquery-rails (4.0.3 > 3.1.2)
  * minitest-rails (2.1.1 > 1.0.0) Gemfile specifies "~> 1.0"
  * minitest-rails-capybara (2.1.1 > 1.0.0)
  * actionmailer (4.2.0 > 4.0.13)
  * actionpack (4.2.0 > 4.0.13)
  * activemodel (4.2.0 > 4.0.13)
  * activerecord (4.2.0 > 4.0.13)
  * activesupport (4.2.0 > 4.0.13)
  * arel (6.0.0 > 4.0.2)
  * bootstrap-wysihtml5-rails (0.3.3.1 > 0.3.2.100)
  * builder (3.2.2 > 3.1.4)
  * capybara (2.4.4 > 2.1.0)
  * hike (2.1.3 > 1.2.3)
  * minitest (5.5.0 > 4.7.5)
  * minitest-capybara (0.7.1 > 0.5.0)
  * minitest-metadata (0.6.0 > 0.5.3)
  * rack (1.6.0 > 1.5.2)
  * rails (4.2.0 > 4.0.13)
  * railties (4.2.0 > 4.0.13)
  * responders (2.0.2 > 1.1.2)
  * sass-rails (5.0.1 > 5.0.0)
  * slop (4.0.0 > 3.6.0)
  * tilt (2.0.1 > 1.4.1)
  * tzinfo (1.2.2 > 0.3.42)
```

Many of these will update alongside Rails, but others are constrained by the Gemfile versioning. My question is this: What's our reasoning for restricting these versions? Can I go ahead and assume, for instance, that our app won't explode if I jump from `database_cleaner` 1.0.1 to 1.4.0? In a similar vein, can I update Rails to 4.2, now that we support it in Cardboard-backed apps?

### CODE REVIEW

**Comment by elfassy** [comment]
any reason we should not be backward compatible with all Rails 4.0 and 4.1?

**Comment by iandonovan** [comment]
Yeah, that's a good point. Of the list above, are there any that we know for sure we can upgrade? I'm also going to float the idea of dropping Rails 4.0 support out there -- it's pretty old, has some deprecated tech, etc. I'm not married to it though.

**Comment by joeyjoejoejr** [comment]
Yeah, I think Ian has a point.  We can probably drop 4.0 support.

**Comment by elfassy** [comment]
"it's pretty old" does not seem to me like a valid argument. Some people might want to use cardboard with Rails 4.0 projects without wanting to upgrade. I would leave it unless we've made a change that is not compatible with that version.
We can bump up all the testing dependencies (minitest, database_cleaner etc) if you want. We just need to make sure tests still run.

**Comment by iandonovan** [comment]
```
In keeping with our maintenance policy, after the upcoming release of 4.2.0, the 4.0 series will be retired. It will not receive further updates for either bug fixes or security issues. All users are urged to migrate to 4.1 as soon as possible.
```
http://weblog.rubyonrails.org/2014/9/12/Rails-4-1-6-and-4-0-10-has-been-released/

I think, if we leave support for 4.0., we need to at least keep its deprecation on the horizon.

**Comment by iandonovan** [comment]
Here's where we're at, as of the most recent commit:
```
Outdated gems included in the bundle:
  * jquery-rails (4.0.3 > 3.1.2)
  * minitest-rails (2.1.1 > 1.0.1)
  * minitest-rails-capybara (2.1.1 > 1.0.0)
  * actionmailer (4.2.0 > 4.0.13)
  * actionpack (4.2.0 > 4.0.13)
  * activemodel (4.2.0 > 4.0.13)
  * activerecord (4.2.0 > 4.0.13)
  * activesupport (4.2.0 > 4.0.13)
  * arel (6.0.0 > 4.0.2)
  * builder (3.2.2 > 3.1.4)
  * capybara (2.4.4 > 2.1.0)
  * hike (2.1.3 > 1.2.3)
  * minitest (5.5.0 > 4.7.0)
  * minitest-capybara (0.7.1 > 0.5.0)
  * minitest-metadata (0.6.0 > 0.5.3)
  * rack (1.6.0 > 1.5.2)
  * rails (4.2.0 > 4.0.13)
  * railties (4.2.0 > 4.0.13)
  * responders (2.0.2 > 1.1.2)
  * sass-rails (5.0.1 > 5.0.0)
  * slop (4.0.0 > 3.6.0)
  * tilt (2.0.1 > 1.4.1)
  * tzinfo (1.2.2 > 0.3.42)
```

File: Gemfile.lock [hunk 1]
```diff
@@ -4,7 +4,7 @@ PATH
     cardboard_cms (0.5.0)
       bootstrap-datepicker-rails
       bootstrap-sass (~> 3.2)
-      bootstrap-wysihtml5-rails (~> 0.3.2.100)
+      bootstrap-wysihtml5-rails (>= 0.3.2.100)
```
Comment (elfassy) [inline Gemfile.lock]: i think this is not a good idea as newer versions might break how we use this gem. How about `~> 0.3.2`

Comment (iandonovan) [inline Gemfile.lock]: Sounds good to me.

**Pull request merged**